### ISSUE TITLE
Wrong automatic code indentation

### ISSUE THREAD

**Issue by cithukyaw** [adobe/brackets#11067]
I have a `switch/case` statement. The code block under a `case` statement has one nested level like below:

```
case 1:
    if ($a) foo();
    break;
```

I wanted to add  curly braces to the `if` statement line. When I typed `{`, the code indentation was automatically moved backwards to the same level of `case` such

```
case 1:
if ($a) {
    foo();
}
    break;
```

Then, I had to manually fix the indentation to get it right like below:

```
case 1:
    if ($a) {
        foo();
    }
    break;
```

To get more clear picture, check the following animated gif or [the original video file](http://im.ezgif.com/tmp/d432815429.mp4) (if it is not broken).

![code-indentation-glitch-08-05-2015-05-33-12](https://cloud.githubusercontent.com/assets/4285555/7535510/f4e6e610-f5ab-11e4-90f3-a09ff7cec017.gif)

I'm not sure it is related to Brackets core or any particular extension. My Brackets version is 1.3.

**Comment by prafulVaishnav** [comment]
@cithukyaw What file you are working on (like JavaScript or other) If you want to make sure its not caused by any extension, reload Brackets without Extension (Debug > Reload Without Extensions)

**Comment by cithukyaw** [comment]
@prafulVaishnav The file is PHP. I reloaded Brackets without Extension. The problem still persists. So, I believe it is an issue of Brackets core.

**Comment by cithukyaw** [comment]
I believe the problem is not particular to the `switch/case` statement only. I noticed that the same problem occurred with the parent `if` or any other control structures.

**Comment by MarcelGerber** [comment]
I can repro in [the CodeMirror php mode](http://codemirror.net/mode/php/), so marking codemirror.
Just to be clear: You've manually indented the if block to that level, correct? Because for me, it's at the same level the `case` is, at first.

**Comment by cithukyaw** [comment]
@MarcelGerber 
> You've manually indented the if block to that level, correct?

Yes, I manually indented.

**Comment by MarcelGerber** [comment]
Upstream bug: codemirror/CodeMirror#3252

**Comment by MarcelGerber** [comment]
This is fixed in the upcoming 1.4 release. The PHP mode's `switch/case` statements now behave the same way they do in JS.
Thanks for reporting!
